rfd/0080-piv-private-key.md

```diff
@@ -0,0 +1,392 @@
+
+It may be possible to work around this limitation by introducing a local proxy to connect to the hardware key, or by supporting a hardware key solution which doesn't need a direct connection, but this is out of scope and will not be explored in this PR. 
+
+In cases where WebUI access is needed or desired, cluster admins should only apply `require_session_mfa: hardware_key | hardware_key_touch` selectively to roles which warrant more protection. Teleport Connect will also serve as a great UI alternative.
+
+### UX
+
+#### Hardware key login
+
+When possible, hardware key login will not be any different from the normal login flow. However, in some cases, additional user intervention will be required. Below are some examples along with the resulting UX.
+
+Note: Teleport Connect will need custom solutions for these edge cases, such as tsh-initiated callbacks.
+ 
+##### Initial login fails due to an unmet private key policy
+
+```
+> tsh login --user=dev
+Enter password for Teleport user dev:
+Tap any security key
+Initial login failed due to an unmet private key policy, "hardware_key".
+Re-initiaiting login with YubiKey generated private key...
+Enter password for Teleport user dev:
+Tap any security key
+```
+
+Note: this should only occur when a user's role determines it's private key policy requirement, and the user does not have an existing login session which meets the required policy (expired or active).
+
+##### User's YubiKey not connected during login
+
+```
+> tsh login --user=dev
+Cluster "root" requires a YubiKey generated private key to login, but there
+is no YubiKey connected. Please insert a YubiKey to re-initiate login...
+// tsh polls the PIV library until the user connects a YubiKey (30 second timeout) or the user cancels
+Re-initiaiting login with YubiKey generated private key.
+Enter password for Teleport user dev:
+Tap any security key
+```
+
+##### User's Yubikey not connected during a request
+
+```
+> tsh ls
+Please insert the YubiKey used during login (serial number XXXXXX) to continue...
+// tsh polls the PIV library until the user connects a YubiKey (30 second timeout) or the user cancels
+```
+
+##### Touch requirement
+
+If `private_key_policy: hardware_key_touch` is required for a user, then Teleport client requests will require touch (cached for 15 seconds). This will be handled by a touch prompt similar to the one used for MFA. This prompt will occur before prompting for login credentials.
+
+```
+> tsh login --user=dev
+Tap your YubiKey
```

Comment:
Do you think it's worth calling out the serial number here, or is it too clunky?